aztec/src/main/kotlin/org/wordpress/aztec/watchers/event/sequence/known/space/API26InWordSpaceInsertionEvent.java

```diff
@@ -0,0 +1,147 @@
+        text length is longer by 1, and the item that is now located at the first BEFORETEXTCHANGED is a SPACE character.
+
+         */
+        if (this.getSequence().size() == sequence.size()) {
+
+            // populate data in our own sequence to be able to run the comparator checks
+            if (!isUserOperationPartiallyObservedInSequence(sequence)) {
+                return false;
+            }
+
+            // ok all events are good individually and match the sequence we want to compare against.
+            // now let's make sure the BEFORE / AFTER situation is what we are trying to identify
+            TextWatcherEvent firstEvent = sequence.get(0);
+            TextWatcherEvent lastEvent = sequence.get(sequence.size()-1);
+
+            // if new text length is longer than original text by 1
+            if (firstEvent.getBeforeEventData().getTextBefore().length() ==
+                    lastEvent.getAfterEventData().getTextAfter().length() - 1) {
+                // now check that the inserted character is actually a space
+                BeforeTextChangedEventData data = firstEvent.getBeforeEventData();
+                if (lastEvent.getAfterEventData().getTextAfter().charAt(data.getStart() + data.getCount()) == SPACE) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean isUserOperationPartiallyObservedInSequence(@NotNull EventSequence<TextWatcherEvent> sequence) {
+        for (int i = 0; i < sequence.size(); i++) {
+
+            TextWatcherEvent eventHolder = getSequence().get(i);
+            TextWatcherEvent observableEvent = sequence.get(i);
+
+            // if time distance between any of the events is longer than 50 millis, discard this as this pattern is
+            // likely not produced by the platform, but rather the user.
+            // WARNING! When debugging with breakpoints, you should disable this check as time can exceed the 50 MS limit and
+            // create undesired behavior.
+            if (i > 0) { // only try to compare when we have at least 2 events, so we can compare with the previous one
+                long timestampForPreviousevent = sequence.get(i-1).getTimestamp();
+                long timeDistance = observableEvent.getTimestamp() - timestampForPreviousevent;
+                if (timeDistance > MAXIMUM_TIME_BETWEEN_EVENTS_IN_PATTERN_MS) {
+                    return false;
+                }
+            }
+
+            eventHolder.setBeforeTextChangedEvent(observableEvent.getBeforeEventData());
+            eventHolder.setOnTextChangedEvent(observableEvent.getOnEventData());
+            eventHolder.setAfterTextChangedEvent(observableEvent.getAfterEventData());
+
+            // items in this.getSequence() should always be ITextWatcherEventComparator,
+            // but we check to avoid any potential problems
+            if (!(eventHolder instanceof ITextWatcherEventComparator
```

Comment:
I wonder if there is way to avoid/make this check unnecessary. Should not all of our TextWatcher events be comparable? Is an external Interface necessary?

Reply:
maybe you're right, we could make these abstract methods of the base `TextWatcherEvent` class so to make subclasses always implement that. Will check and get back here.

Reply:
Addressed the elimination of the interface in 91f3edc - didn't make the class abstract as this is a little different situation where we're not testing `TextWatcherEvent`s against each other but rather asking the event itself to look into its inner data and determine if the data collected corresponds to its class. 

Some considerations about it: a `TextWatcherEvent` object alone doesn't necessarily know anything of this as it just holds data. But a `TextWatcherEventDeleteText` or a `TextWatcherEventInsertText` should be able to determine, by looking into the data being held within, whether this is an insertion or a deletion action happening. In this way, as we are describing an _ability_  these two classes have (but not the base class) it is that I thought an `implements interface` kind of relationship would be better suited.

I think the interface makes this a bit clearer that it is now after commit 91f3edc, but let me know your thoughts and we can either leave it as is now (with the change) or revert if you consider the prior looked better. I'd vote for the interface, but wanted to give it a try and check what you think as well 👍 - let me know your thoughts 🙇

Reply:
Thanks Mario for taking in consideration my comments and replied with details. There is only one thing left :)
The comment in the code here https://github.com/wordpress-mobile/AztecEditor-Android/commit/91f3edc61da9d8aa6c98171ff8c6e5fa93819526#diff-faa8a86eaccb01070a180da4df06e11eR109 that still reference the Interface

Reply:
👍 addressed in 3b57fdc